pyERGM/metrics.py

```diff
@@ -727,6 +727,42 @@ def __str__(self):
         return "node_attribute_in"
 
 
+class SumDistancesConnectedNeurons(ExWeightNumEdges):
+    """
+    This function calculates the sum of euclidean distances between all pairs of connected neurons.
+    rows = samples, columns = axes
+    """
+
+    def __init__(self, exogenous_attr: Collection, is_directed: bool):
+        # I am not sure what is the meaning of defining exogenous_attr as collection, I just saw it was done
+        # in other functions inheriting from ExWeightNumEdges. is there a way to specify that the wanted dtypes are:
+        # pd.DataFrame | pd.Series | np.ndarray?
+#
+        # check inputs
+        if isinstance(exogenous_attr, (pd.DataFrame, pd.Series)):
+            exogenous_attr = exogenous_attr.to_numpy(copy=True)
+        elif not isinstance(exogenous_attr, np.ndarray):
+            raise ValueError(f"Unsupported type of positions: {type(exogenous_attr)}. Supported types are pd.DataFrame, "
+                             f"pd.Series and np.ndarray.")
+        if len(exogenous_attr.shape) == 1:
+            exogenous_attr = exogenous_attr.reshape(-1, 1)
+
+        super().__init__(exogenous_attr)
+        self._is_directed = is_directed
+
+    def _calc_edge_weights(self):
+        num_nodes = len(self.exogenous_attr)
+        self.edge_weights = np.reshape(distance.squareform(distance.pdist(self.exogenous_attr, metric='euclidean')), (self._get_num_weight_mats(), num_nodes, num_nodes))
+
+    def _get_num_weight_mats(self):
+        return 1
+    # wasn't sure what this is for. I understood the extra dimension is required for the einsum in calculate,
```

Comment:
There may be cases where metrics have more than one feature they calculate, and then more than a single weight matrix is needed (e.g., if we wanted to sum the distances between different sets of neurons, something like a merging of distances and types, we could hole multiple weight matrices, where in each matrix only the relevant entries are filled and the rest are 0, so they won't sum up). 
Originally this class covered more exogenous metrics (specifically, number of edges between types), and this was necessary there. We changed the implementation of the types metric for runtime considerations. Now there is no example for a usage of this multiple-feature calculation, but we support it.

Reply:
ok thank you!